### ISSUE TITLE
So, the code seems to optimize potenital by deep learning, how can we use the property of implicit regularization of mirror descent in it, i.e. argminD_\fai(w,w_0)

### ISSUE THREAD

**Issue by XX824** [alexsemendinger/mirror-descent#1]
So, the code seems to optimize potenital by deep learning, how can we use the property of implicit regularization of mirror descent in it, i.e. argminD_\fai(w,w_0)

**Comment by XX824** [comment]
Can the author discuss the issue with me? Thanks a lot!

**Comment by alexsemendinger** [comment]
Hi! I'm not currently moving forward with this project but I'd be happy to discuss with you.

The code doesn't involve deep learning / neural networks per se. Rather, it's meant as a proof of concept of a certain two-level optimization process (which, if useful, might be applied to neural networks).

**Comment by XX824** [comment]
Happy to recieve the answer from you! But from the article that you cited'[Data-Driven Mirror Descent in Input-Convex Neural Networks](https://arxiv.org/pdf/2206.06733.pdf)'. It seems that the project is devoted to learn a potential which could have a better generalization. My question is that how can we build a connection between the learning of potential and the property of mirror descent, i.e. argminD_\fai(w,w_0). From the roadmap that you've written, my understanding is that you are going to study the relationship between the \omega_star and the optimal potential. And I wonder if the property that argminD_\fai(w,w_0) will help or not? Actually, I don't get what you mean from your comment above. Hope to receive from you!

**Comment by alexsemendinger** [comment]
This repo is essentially working notes for a project I'm not pursuing anymore, so it's not meant to be particularly polished or to answer any particular questions. (I'm curious how you came across it and what you're hoping to take from it -- because it's fairly incomplete, I'm not sure what use it is to others in its current state. But if you find something useful then I'm glad!)

The overall idea is something like "in what settings might it make sense to use a learned optimization algorithm (e.g. to obtain favorable inductive biases), and how might we implement that?"

This code is something like step 1 (maybe step 0) of that project: implementing a very simple version of an optimization algorithm with some parameters that can be optimized (mirror descent / preconditioned gradient descent). This is also simple enough to analyze by hand, so we could derive an expression for what potential we'd expect to learn and check that the code finds this.

Verifying that the two-level optimization works and that the learned potential $Q^*$ matched my calculations is about as far as these notebooks get (plus some additional poking around at the solution). While there may be some way to bridge the (large) gap between this work and the larger questions about neural networks, I moved on from this project partly because I didn't have a clear picture myself of what that was.

I'm going to close this issue, but if you want to chat further you can reach me at alex.semendinger at gmail.
